dirty_cat/datasets/_ken_embeddings.py

```diff
@@ -65,36 +65,32 @@ def get_ken_embeddings(
     # Get all embeddings:
     emb_full = fetch_figshare(emb_id)
     emb_type = fetch_figshare(emb_type_id)
-    emb_full = pd.read_parquet(emb_full["path"])
     emb_type = pd.read_parquet(emb_type["path"])
     # All in lower case for easier matching
     emb_type["Type"] = emb_type["Type"].str.lower()
-    emb_type = emb_type[emb_type["Type"].str.contains(types)].drop_duplicates(
-        subset=["Entity"]
-    )
+    emb_type = emb_type[emb_type["Type"].str.contains(types)]
+    emb_type.drop_duplicates(subset=["Entity"], inplace=True)
     if exclude is not None:
         emb_type = emb_type[~emb_type["Type"].str.contains(exclude)]
-    emb_extracts = emb_type.join(emb_full.set_index("Entity"), on="Entity")
-    del emb_full, emb_type
-    emb_extracts["Entity"] = (
-        emb_extracts["Entity"]
-        .str.replace("<", "")
-        .str.replace(">", "")
-        .str.replace("_", " ")
-    )
-    emb_extracts.reset_index(drop=True, inplace=True)
-    if pca_components is not None:
-        pca_i = PCA(n_components=pca_components, random_state=0)
-        emb_columns = []
-        for j in range(pca_components):
-            name = "X" + str(j) + suffix
-            emb_columns.append(name)
-        pca_embeddings = pca_i.fit_transform(
-            emb_extracts.drop(columns=["Entity", "Type"])
-        )
-        pca_embeddings = pd.DataFrame(pca_embeddings, columns=emb_columns)
-        emb_pca = pd.concat([emb_extracts[["Entity", "Type"]], pca_embeddings], axis=1)
-        del pca_embeddings
-        return emb_pca
-    else:
-        return emb_extracts
+    emb_final = pd.DataFrame()
+    for path in emb_full["path"]:
+        emb_extracts = pd.read_parquet(path)
+        emb_extracts = pd.merge(emb_type, emb_extracts, on="Entity")
+        emb_extracts.reset_index(drop=True, inplace=True)
+        if pca_components is not None:
+            pca_i = PCA(n_components=pca_components, random_state=0)
+            emb_columns = []
+            for j in range(pca_components):
+                name = "X" + str(j) + suffix
+                emb_columns.append(name)
+            pca_embeddings = pca_i.fit_transform(
+                emb_extracts.drop(columns=["Entity", "Type"])
+            )
+            pca_embeddings = pd.DataFrame(pca_embeddings, columns=emb_columns)
+            emb_pca = pd.concat(
+                [emb_extracts[["Entity", "Type"]], pca_embeddings], axis=1
+            )
+            emb_final = pd.concat([emb_final, emb_pca])
+        else:
+            emb_final = pd.concat([emb_final, emb_extracts])
```

Comment:
I'm not very good at pandas, but my gut feeling is that growing a pandas dataframe like this might lead to a duplication of `emb_final` at each iteration.

I would try assembling the DataFrame at the end. To get a good understanding of the tradeoffs, it could be useful to do a small script to compare different ways of assembling the data in a DataFrame and profile it, including using memprofiler.

Reply:
Thank you, this is great!
I will look for more improvements of this type so that we may avoid the memory errors.

Reply:
Do we have an idea of how far above the memory limit we are?

Reply:
Not sure, still trying to find. I know the resources allocated by circleci are 2 CPUs and 4GB of RAM.

Reply:
![Figure_1](https://user-images.githubusercontent.com/62058944/218975259-59fed3fd-1fbd-4a4a-ae54-562cb2938b60.png)

Ok, just did a memory profiling and it seems that we are around 1GB above the limit at one point.

Reply:
![Screenshot from 2023-02-15 09-44-35](https://user-images.githubusercontent.com/62058944/218977362-e0c40953-4b62-4212-ad0c-b76338a98cb6.png)
Of course, it's the first get_ken_embeddings function (when getting all games) that is the cause. 
But interestingly, when repeated with line by line analysis we are somewhat under the limit.